src/main/scala/strawman/collection/View.scala

```diff
@@ -290,22 +287,93 @@ trait ArrayLike[+A] extends Any {
   def apply(i: Int): A
 }
 
+trait SeqView[+A] extends View[A] with SeqOps[A, SeqView, SeqView[A]] { self =>
+  override def view: SeqView[A] = this
+
+  override def toSeq = immutable.IndexedSeq.from(this)
+
+  override def iterableFactory: IterableFactoryLike[SeqView] = SeqView
+
+  override protected[this] def fromSpecificIterable(coll: Iterable[A]): SeqView[A] = fromIterable(coll)
+
+  override protected[this] def newSpecificBuilder(): Builder[A, SeqView[A]] =
+    IndexedSeq.newBuilder().mapResult(_.view)
+
+  override def toString = "SeqView(?)"
+  override def className = "SeqView"
+
+  override def apply(i: Int) = iterator().drop(i - 1).next()
+  override def prepended[B >: A](elem: B): SeqView[B] = SeqView.Prepend(elem, this)
+  override def appended[B >: A](elem: B): SeqView[B] = SeqView.Append(this, elem)
+  override def prependedAll[B >: A](prefix: Iterable[B]): SeqView[B] = SeqView.PrependAll(prefix, this)
+  override def appendedAll[B >: A](suffix: Iterable[B]): SeqView[B] = SeqView.AppendAll(this, suffix)
+}
+
+object SeqView extends SeqFactory[SeqView] {
+  def fromIterator[A](it: => Iterator[A]): SeqView[A] = new SeqView[A] {
+    override def length = it.size
```

Comment:
That looks wrong

Reply:
Indeed, I guess we could buffer `it` when `size` is called (and use the buffer for any succequent `apply` or `iterator` calls, or we could simply remove `fromIterator` from the `SeqView` companion. Which do you prefer?

Reply:
I have replaced `SeqView.fromIterator` with `SeqView.fromIteratorProvider` and `IndexedSeqView` with `IndexedSeqView.fromIteratorProvider` in commits 8b3aabb and de8ab24 respectively.